File: src/logging.m
#import "logging.h"
#import <stdarg.h>
#include <stdio.h>

static BOOL verboseEnabled = NO;

void LogSetVerbose(BOOL enabled) {
    verboseEnabled = enabled;
}

static void logMessage(FILE *stream, const char *level, NSString *format, va_list args) {
    if (!format || !stream || !level) {
        return;
    }
    
    NSString *message = [[NSString alloc] initWithFormat:format arguments:args];
    if (!message) {
        return;
    }
    
    fprintf(stream, "lidoff[%s]: %s\n", level, message.UTF8String);
    fflush(stream);
}

void LogInfo(NSString *format, ...) {
    va_list args;
    va_start(args, format);
    logMessage(stdout, "info", format, args);
    va_end(args);
}

void LogError(NSString *format, ...) {
    va_list args;
    va_start(args, format);
    logMessage(stderr, "error", format, args);
    va_end(args);
}

void LogDebug(NSString *format, ...) {
    if (!verboseEnabled) {
        return;
    }
    
    va_list args;
    va_start(args, format);
    logMessage(stdout, "debug", format, args);
    va_end(args);
}

File: src/logging.rs
use std::sync::atomic::{AtomicBool, Ordering};

static VERBOSE_ENABLED: AtomicBool = AtomicBool::new(false);

pub fn set_verbose(enabled: bool) {
    VERBOSE_ENABLED.store(enabled, Ordering::Relaxed);
}

pub fn info(message: impl AsRef<str>) {
    println!("lidoff[info]: {}", message.as_ref());
}

pub fn error(message: impl AsRef<str>) {
    eprintln!("lidoff[error]: {}", message.as_ref());
}

pub fn debug(message: impl AsRef<str>) {
    if VERBOSE_ENABLED.load(Ordering::Relaxed) {
        println!("lidoff[debug]: {}", message.as_ref());
    }
}
